### DISCUSSION TITLE
Database size and performance on Pi models

### DISCUSSION THREAD

**Discussion by qurm** [SBFspot/SBFspot#479] [Q&A] [answered]
Hi all,

I have been running SBFSpot for about 4 years, since 2018, with a single system and default values (readings every 5m).  This has resulted in a current SQLite database of 30MB.

This running on a Raspberry pi zero w rev 1.1  (512MB, 1Ghz).

Generally has been running OK, but I am seeing that it is very slow when I `ssh` into it and at times I see that the SBFspotUploadDaemon is taking high CPU  (say 80-90%), seen in `top`.  Does this seem reasonable performance?  What do others see on a Pi Zero?

I recently had some SD card issues, and this is a fresh build with Raspbian Buster on a 32GB card. so I'd expect the system is otherwise healthy.

Thanks, Andy

**Comment by sbf-** [comment, marked as answer]
This is a known problem and i’m still working to include [this workaround]( https://github.com/SBFspot/SBFspot/issues/283#issuecomment-541570027 ) into the official release. It’s a matter of time.

**Reply by qurm** [reply]
Thanks - will review that and try it.

**Reply by qurm** [reply]
I have applied the db view changes and rebuilt the sbfspot source with that SQL change.  Performance does seem better thanks, but still seeing similar high CPU.  I'll let it run for a day or two and check the logs, then raise an issue?   Thanks for your help!
